### DISCUSSION TITLE
unexpected error

### DISCUSSION THREAD

**Discussion by Sweetheartman** [minio/minio#12840] [Q&A] [answered]
An error is reported when building a MinIO cluster based on RAMDisk, but the stand-alone version based on RAMDisk can be used

Error: disk returned an unexpected error at http://10.196.3.188:9001/mnt/ramdisk/data2, please investigate - disk will be offline (open /mnt/ramdisk/data2/.minio.sys/tmp/.writable-check-2a4d483d775457a3.tmp: invalid argument) (*fmt.wrapError)

RAMDisk：
mount -t tmpfs -o rw,size=2G tmpfs /mnt/ramdisk

**Comment by vadmeste** [comment, marked as answer]
@Sweetheartman MinIO requires O_DIRECT support from disks because tmpfs doesn't support that.

**Comment by Sweetheartman** [comment]
> @vadmeste  MinIO requires O_DIRECT support from disks because tmpfs doesn't support that.
However, why is it possible to use RAMDisk in stand-alone mode, but not in cluster mode?

**Comment by harshavardhana** [comment]
> > @vadmeste  MinIO requires O_DIRECT support from disks because tmpfs doesn't support that.
> > However, why is it possible to use RAMDisk in stand-alone mode, but not in cluster mode?

Single drive mode doesn't require O_DIRECT
